src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -0,0 +1,429 @@
+                { 11920.69f, -7061.134f, 30.40684f },
+                { 11920.56f, -7067.133f, 30.41605f },
+                { 11920.76f, -7058.135f, 30.61656f },
+                { 11920.5f, -7070.132f, 30.40836f }
+        };
+
+Position const thalorienSummon = { 11795.32f, -7070.476f, 26.27511f, 5.67232f  };
+Position const thalorienFight = { 11788.46f, -7063.375f, 25.79677f, 3.054326f };
+Position const morlenSummon = { 11766.46f, -7050.078f, 26.19846f, 5.637414f  };
+
+class npc_thalorien_dawnseeker : public CreatureScript
+{
+public:
+    npc_thalorien_dawnseeker() : CreatureScript("npc_thalorien_dawnseeker") { }
+
+    struct npc_thalorien_dawnseekerAI : public ScriptedAI
+    {
+        npc_thalorien_dawnseekerAI(Creature* creature) : ScriptedAI(creature), _summons(me) { }
+
+        bool GossipSelect(Player* player, uint32 menuId, uint32 /*gossipListId*/) override
+        {
+            if (menuId == MENU_EXAMINE_REMAINS)
+            {
+                CloseGossipMenuFor(player);
+                DoAction(ACTION_START_QUEST);
+                _playerGUID = player->GetGUID();
+            }
+            return false;
+        }
+
+        void DoAction(int32 action) override
+        {
+            if (action == ACTION_START_QUEST)
+            {
+                me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+
+                if (Creature* thalorien = me->SummonCreature(NPC_THALORIEN_DAWNSEEKER, thalorienSummon, TEMPSUMMON_CORPSE_DESPAWN, 0))
+                    _thalorienGUID = thalorien->GetGUID();
+                me->SummonCreatureGroup(SUMMON_SUNWELL_DEFENDER);
+
+                _events.ScheduleEvent(EVENT_INTRO_0, 3s);
+            }
+        }
+
+        void Reset() override
+        {
+            me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+            _summons.DespawnAll();
+            _playerGUID.Clear();
+            _morlenGUID.Clear();
+            _thalorienGUID.Clear();
+            _events.Reset();
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_INTRO_0:
+                        if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+                            thalorien->AI()->Talk(SAY_THALORIEN_1);
+
+                        _events.ScheduleEvent(EVENT_INTRO_1, 5s);
+                        break;
+                    case EVENT_INTRO_1:
+                        if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
+                            thalorien->AI()->Talk(SAY_THALORIEN_2);
+
+                        _events.ScheduleEvent(EVENT_INTRO_2, 6s);
+                        break;
+                    case EVENT_INTRO_2:
+                        if (Creature* thalorien = ObjectAccessor::GetCreature(*me, _thalorienGUID))
```

Comment:
Create a function for those calls maybe, that's a lot of code duplication ;p
void ThalorienTalk(uint32 talkId)
{
    if (Creature* thalorien blablabla........)
        Talk(talkId);
}